proparse/src/main/java/org/prorefactor/core/ProToken.java

```diff
@@ -171,4 +184,29 @@ public String toString() {
         + ",file=" + fileIndex + ":" + endFile + ",line=" + line + ":" + endLine + ",col=" + col + ":" + endColumn
         + "]";
   }
+
+  /**
+   * @return True if token is part of an editable section in AppBuilder managed code
+   */
+  public static boolean isEditableInAB(String str) {
+    if (str == null)
+      return true;
+    List<String> attrs = Splitter.on(',').omitEmptyStrings().trimResults().splitToList(str);
+    if (attrs.isEmpty() || !"_UIB-CODE-BLOCK".equalsIgnoreCase(attrs.get(0)))
+      return false;
+
+    if ((attrs.size() >= 3) && "_CUSTOM".equalsIgnoreCase(attrs.get(1))
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this conditional structure or edit its code blocks so that they're not all the same. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=squid%3AS3923)